WordPress/src/main/java/org/wordpress/android/ui/jetpack/backup/details/BackupDownloadDetailsViewModel.kt

```diff
@@ -1,15 +1,97 @@
 package org.wordpress.android.ui.jetpack.backup.details
 
-import androidx.lifecycle.ViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.wordpress.android.R
+import org.wordpress.android.modules.BG_THREAD
+import org.wordpress.android.modules.UI_THREAD
+import org.wordpress.android.ui.jetpack.BackupAvailableItemsProvider
+import org.wordpress.android.ui.jetpack.BackupAvailableItemsProvider.BackupAvailableItem
+import org.wordpress.android.ui.jetpack.BackupAvailableItemsProvider.BackupAvailableItemType
+import org.wordpress.android.ui.jetpack.backup.details.BackupDownloadDetailsViewModel.UiState.Content
+import org.wordpress.android.ui.utils.UiString
+import org.wordpress.android.ui.utils.UiString.UiStringRes
+import org.wordpress.android.viewmodel.ScopedViewModel
 import javax.inject.Inject
+import javax.inject.Named
 
-class BackupDownloadDetailsViewModel @Inject constructor() : ViewModel() {
+class BackupDownloadDetailsViewModel @Inject constructor(
+    private val backupAvailableItemsProvider: BackupAvailableItemsProvider,
+    @Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher,
+    @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher
+) : ScopedViewModel(mainDispatcher) {
     private var isStarted: Boolean = false
+
+    private val _uiState = MutableLiveData<UiState>()
+    val uiState: LiveData<UiState> = _uiState
+
     fun start() {
-        if (isStarted) {
-            return
+        if (isStarted) return
+        isStarted = true
+
+        // todo: annmarie - this is a temp name until useCase has been added for record dets :)
+        getData()
+    }
+
+    private fun getData() {
+        launch {
+            val availableItems = backupAvailableItemsProvider.getAvailableItems()
+            _uiState.value = buildContentUiState(availableItems)
         }
+    }
 
-        isStarted = true
+    private suspend fun buildContentUiState(items: List<BackupAvailableItem>): Content {
+        return withContext(bgDispatcher) {
+            val availableItemsListItems: List<ListItemUiState> = items.map {
+                ListItemUiState(
+                        availableItemType = it.availableItemType,
+                        label = UiStringRes(it.labelResId),
+                        checked = true,
+                        onClick = { onItemClicked(it.availableItemType) }
+                )
+            }
+            // todo: annmarie - swap out the placeholder for date from record
+            Content(
+                    description = UiStringRes(R.string.backup_download_details_description),
+                    items = availableItemsListItems
```

Comment:
Question (❓ ): How would the `1%1s` be replaced in this context?

Reply:
Not sure how I am going to handle this - that's why I left the comment 😉